project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -351,6 +353,36 @@ public void testJBallerinaBackend() {
         Assert.assertEquals(platformLibraries.size(), 4);
     }
 
+    @Test(description = "tests langlib files are not included in the jar file collection for executing the test")
+    public void testLanglibFilesAreNotIncluded() {
+        Path projectPath = tempResourceDir.resolve("myproject_with_lang_import");
+
+        // 1) Initialize the project instance
+        BuildOptions buildOptions = BuildOptions.builder().setSkipTests(false).build();
+        BuildProject project = loadBuildProject(projectPath, buildOptions);
+        // 2) Compile the current package
+        PackageCompilation compilation = project.currentPackage().getCompilation();
+        JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(compilation, JvmTarget.JAVA_17);
+
+        // 3) get topologically sorted modules
+        for(ModuleDescriptor moduleDescriptor :
+                project.currentPackage().moduleDependencyGraph().toTopologicallySortedList()) {
+
+            // 4) get jar files for each module
+            Collection<JarLibrary> jarLibraries = jBallerinaBackend.jarResolver()
+                    .getJarFilePathsRequiredForTestExecution(moduleDescriptor.name());
+
+            // 5) assert that langlib files are not included in the jar files
+            for (JarLibrary jarLibrary : jarLibraries) {
+                boolean found = Pattern.matches(".*ballerina-lang\\..+.jar", jarLibrary.path().toString());
+                if(found) {
```

Comment:
```suggestion
                if (found) {
```

Look for other similar places

Reply:
fixed in ad36dfb102c68ab2de58f775a69101b820e6c168 commit